File: split/endgraf.pas
procedure endgraf;
begin
  if curlist.modefield = 102 then
  begin
    if curlist.headfield = curlist.tailfield then
      popnest
    else
      linebreak(eqtb[5269].int);
    normalparagraph;
    errorcount := 0;
  end;
end;

File: cpp/endgraf.cpp
#include "endgraf.h"
#include "popnest.h"
#include "linebreak.h"
#include "normalparagraph.h"

static int widow_penalty(void) { return int_par(widow_penalty_code); }

void endgraf(void)
{
	if (mode == hmode)
	{
		if (head == tail)
			popnest();
		else
			linebreak(widow_penalty());
		normalparagraph();
		errorcount = 0;
	}
}
